keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertex.java

```diff
@@ -21,7 +23,7 @@
 import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
 import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.PartialDerivatives;
 
-public class GaussianVertex extends DoubleVertex implements ProbabilisticDouble {
+public class GaussianVertex extends DoubleVertex implements ProbabilisticDouble, ProposalVertex {
```

Comment:
I like that the GaussianVertex can be used easily with KLDivergence but ideally this feature would be decoupled from the GaussianVertex itself. Can we add a `compute(ProbabilisticDouble q, NetworkSamples samples)` that does the wrapping inside the KLDivergence class?